Assets/Scripts/Dummy/LiveTarget.cs

```diff
@@ -0,0 +1,66 @@
+using UnityEngine;
+
+namespace Dummy
+{
+    public class LiveTarget : MonoBehaviour
+    {
+        public bool isInitiallyAlive;
+        public bool removeOnDeath;
+    
+        [Min(0)]
+        public float removeDelay;
+        public bool IsAlive
+        {
+            get => _isAlive;
+            set
+            {
+                _isAlive = value;
+                foreach (var rb in GetComponentsInChildren<Rigidbody>())
```

Comment:
Generally, it is a bad practice to put so much heavy logic inside a property. Use methods for that